### ISSUE TITLE
How to visualize the point clouds as the same way in paper?

### ISSUE THREAD

**Issue by mtli77** [charusharma1991/SSL_PointClouds#1]
Hi @charusharma1991 

Thank you for sharing the wonderful project! 
I am very interested in the visualization of the point clouds in your paper, could you consider to release the visualization code?
And also, I am confused about the data process and the few shot training on the segmentation task, could you please give some more specific introduction about it?

Waiting for your reply!
Thank you

**Comment by charusharma1991** [comment]
Hi,
Thanks for your interest in our paper!
We've used Mitsuba Renderer for visualization of point clouds. Mitsuba can be easily installed on Mac or Windows from [here](http://www.mitsuba-renderer.org/download.html). It takes .xml file as an input to visualize PCs. I have written the code for .xml files by myself. I will see if I can release the code which can generate .xml files.

Regarding the segmentation task, it is same till SSL part. The only change occurs in the downstream network. You can also refer DGCNN and PointNet papers for the Segmentation network.
Thanks!

**Comment by mtli77** [comment]
> Hi,
> Thanks for your interest in our paper!
> We've used Mitsuba Renderer for visualization of point clouds. Mitsuba can be easily installed on Mac or Windows from [here](http://www.mitsuba-renderer.org/download.html). It takes .xml file as an input to visualize PCs. I have written the code for .xml files by myself. I will see if I can release the code which can generate .xml files.
> 
> Regarding the segmentation task, it is same till SSL part. The only change occurs in the downstream network. You can also refer DGCNN and PointNet papers for the Segmentation network.
> Thanks!

Hi, @charusharma1991 
During the self-supervision step, the training strategy is to sample the support set from the training dataset, and there is no query set. This seems different from the previous self-supervised strategy of pre-training with other datasets.
I wonder that does this because the dataset used in self-supervision is the same as the dataset with downstream task FSL, so only a part of the dataset must be sampled for self-supervised pre-training?

**Comment by charusharma1991** [comment]
The SSL part is same for both classification and segmentation. You can use a query set in SSL part too but that's only to check the loss.
Yes, we sample a part of the dataset as this follows FSL setup and the sampled support and query set we use in both SSL and FSL are same.
